feat(lightspeed): add attachment support and disclaimers

karthikjeeyar opened the pull request:
## Fixes

https://issues.redhat.com/browse/RHDHPAI-555
https://issues.redhat.com/browse/RHDHPAI-758
https://issues.redhat.com/browse/RHDHPAI-762


This PR adds the following:

- Attachment support from UI. Supported file type - `.yaml`, `.json` and `.txt` files.
- Updated Functional and Legal Disclaimers
- Updated welcome prompts on new conversation page. (Hard coded for now, will make it configurable in followup PR)
- Shows Citation links if the responses contains the information.


<!-- Please describe what you added, and add a screenshot if possible.
     That makes it easier to understand the change so we can :shipit: faster. -->

#### :heavy_check_mark: Checklist

## Screenshots

Functional and Legal Disclaimers:

![image](https://github.com/user-attachments/assets/f2add7b5-66cf-4ebc-82a1-6b152ccfd1ca)

Attachement demo:

https://github.com/user-attachments/assets/edeb5f77-b6c9-49b0-b479-66de01024b7b

Attachment preview:

![image](https://github.com/user-attachments/assets/3466c8e1-498d-405b-b0a6-c34c94de45a6)

Edit attachment:

![image](https://github.com/user-attachments/assets/96eb31f5-6128-4e28-99ad-4402ed6c453c)


Attachment Error states:

<img width="885" alt="info_message" src="https://github.com/user-attachments/assets/d0a85c36-0b78-41c0-958b-1d448d599864" />

---

<img width="880" alt="error_screen" src="https://github.com/user-attachments/assets/eb4b576a-afda-4b43-a192-8db157f03451" />

---

<!--- Please include the following in your Pull Request when applicable: -->

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/redhat-developer/rhdh-plugins/blob/main/CONTRIBUTING.md#creating-changesets))
- [ ] Added or Updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

ShiranHi commented:
Thank you @karthikjeeyar , looks great! 

I have a few suggestions and questions regarding some UI elements:
1. Could we make the "Edit," "Save," and "Cancel" buttons a bit smaller?
2. Would it be possible to change the "Dismiss" button text to "Close"?
3. For failed file uploads, could we provide users with a list of supported file formats?
4. When a file already exists, instead of an alert, could we display an informational message with the title "File already uploaded"?
5. I'm curious about the origin of the current edit behavior. Could you point me to where that implementation came from? I have some ideas on how we might improve it.

cc @aprilma419

karthikjeeyar commented:
Thanks @ShiranHi for the review. I have opened an issue in PF chatbot to address `#1` & `#2`, once it is available we can get this fixed in followup PR and I have addressed `#3` and `#4`, updated the screenshots in the description.

for `#5`, the edit attachment implementation is coming from PF - ref: https://pf-virtual-assistant.surge.sh/patternfly-ai/chatbot/messages/#editable-attachments

yangcao77 commented:
changes look good to me. but with this change https://github.com/redhat-developer/rhdh-plugins/pull/767

according to https://issues.redhat.com/browse/RHDHPAI-563
should show functional disclaimer is the question validation is enabled (by default, or `disableQuestionValidation=false` ), and show warning if `disableQuestionValidation=true`

yangcao77 commented:
> tional disclaimer (static text) in new or existing conversation sessions. I think topic restrictions warning message is a different feature which I have not covered in this PR (we also need some designs for this).
> 
> Update:
> 
> We discussed this topic in our sync meeting, Ben suggested to add a additional disclaimer just below the functional disclaimer, if the platform engineer disables questionValidation. I will update this in this PR.

`disableQuestionValidation` has been set as frontend visible in this PR   https://github.com/redhat-developer/rhdh-plugins/pull/767, so frontend should be able to directly get it from config.
as the disclaimers are now hard-coded in frontend side, should I close the rest of the issues under epic: https://issues.redhat.com/browse/RHDHPAI-563 ?

karthikjeeyar commented:
@yangcao77 After speaking with Rohit today, we decided to add this additional topic restriction warning in a followup PR as this PR already has multiple features and adding more to this may be difficult for reviewing.
